src/views/Puzzle/Puzzle.tsx

```diff
@@ -907,11 +907,30 @@ export class _Puzzle extends React.Component<PuzzleProperties, PuzzleState> {
         const show_correct = this.state.show_correct;
 
         let next_id = 0;
+        let random_id = 0;
+        let temp_id = 0;
         for (let i = 0; i < this.state.puzzle_collection_summary.length - 1; ++i) {
             if (this.state.puzzle_collection_summary[i].id === puzzle.id) {
                 next_id = this.state.puzzle_collection_summary[i + 1].id;
             }
         }
+        if (this.state.puzzle_collection_summary.length > 1) {
+            temp_id = Math.floor(Math.random() * this.state.puzzle_collection_summary.length);
+            random_id = this.state.puzzle_collection_summary[temp_id].id;
+            if (random_id === puzzle.id) {
```

Comment:
If we remove the current puzzle *before* rolling the dice, I think the code will become simpler, and we will keep randomness in more cases.

```
const ids = this.state.puzzle_collection_summary
    .map((p) => p.id)
    .filter((id) => id !== puzzle.id);
const random_id = ids[Math.floor(Math.random()*ids.length)];
```

---

Logically, this here is what I am suggesting - instead of:

1) Select a puzzle at random from the whole group
2) If current puzzle was selected:
   a) If first puzzle, go next
   b) If last puzzle, go back. I'm not sure why this has to be handled separately from (c)
   c) otherwise, go back one

We do:

1) Remove the current puzzle from the list
2) Select a puzzle at random

Reply:
Or keep it truly random and occasionally test whether they remember the solution they just found 😅

Reply:
Got a bit busy - updated this now. Yeah it seems like a nice way to do it :)

Reply:
Feel free to have another look also :)